interface.go

```diff
@@ -47,9 +47,9 @@ type Multiaddr interface {
 	//
 	Encapsulate(Multiaddr) Multiaddr
 
-	// Decapsultate removes a Multiaddr wrapping. For example:
+	// Decapsulate removes a Multiaddr wrapping. For example:
 	//
-	//      /ip4/1.2.3.4/tcp/80 decapsulate /ip4/1.2.3.4 = /tcp/80
+	//      /ip4/1.2.3.4/tcp/80 decapsulate /tcp/80 = /ip4/1.2.3.4
```

Comment:
Should this also document what happens if you try to decapsulate something that's not there at the right hand side of the multiaddr (it returns `nil`).

Reply:
It should! But the behavior is `/ip4/1.2.3.4/tcp/80 decapsulate /udp/80 = /ip4/1.2.3.4/tcp/80`